### ISSUE TITLE
Type mismatch error when trying to create a new page if none exist already

### ISSUE THREAD

**Issue by antiftw** [bolt/core#1734]
**Summary**:

I just rolled out Bolt to a test domain to play around with it, and I wanted to do it without using the data from the fixtures, because I wanted to see what was needed to get it to run without using that. 

I think it is related to the fact there a no existing pages because when I tried it after I ran the fixtures command I had no problems adding a new page.

When trying to add a new page I got a type mismatch error:

Argument 1 passed to Bolt\Twig\ContentExtension::getListTemplates() must be an instance of Bolt\Entity\Field\TemplateselectField, instance of Bolt\Entity\Field given, called in /var/www/bolt/var/cache/prod/twig/b9/b9a603f495ebe0805699952bde34c146cbbcb1453ce1171812c6a0be4681e59e.php on line 45

**Specifics**

The url: 

https://hostname.com/bolt/new/pages

**The version** (output fron bin/console bolt:info):

Bolt version: 4.0.0 RC 33
 
 * Install type: Composer install
 * Database: mysql 5.7.31-0ubuntu0.18.04.1 - Localhost via UNIX socket (with JSON)
 * PHP version: 7.2.24-0ubuntu0.18.04.6
 * Symfony version: v5.1.3
 * Operating System: Linux - 4.15.0-112-generic

**Tried it in:** 

Firefox and Chromium

**Steps to reproduce:**

1) go to /bolt and login to the backend
2) hover over pages in the left menu
3) click on new page in the menu that appears

**Expected result:**

I expected to get the webpage to create a new "page".

**Actual result:**

I got the error as mentioned above and as can be seen in the screenshot:

![screenshot_bolt_newpage](https://user-images.githubusercontent.com/21287634/90621521-1f441400-e214-11ea-984c-f905254592d3.png)

I hope my description was clear enough, since it was my first time reporting a bug for anything on Github. If more details are required, please do not hesitate to ask. I also tried finding a similar issue in the list but could not find it skimming the titles.

Kind regards,
AntiFTW

**Comment by bobdenotter** [comment]
Hi @antiftw. 

I've moved the issue to the correct repo. I'll see if I can reproduce and fix the issue!

**Comment by antiftw** [comment]
Hi @bobdenotter,

Okay, I tried looking for the function call but I am quite inexperienced with Bolt still, and with contributing to Github in general, I figure it probably is an issue related to the migration from 3 -> 4, since similar functionality like adding blocks or creating new entries does work like expected.

Hope you can reproduce it.

**Comment by bobdenotter** [comment]
Hmm, i've not been able yet to reproduce this.. I assume what you did is this, right? 

```
composer create-project bolt/project myprojectname
cd myprojectname
``` 

(then configure `.env` to put in your MySQL db credentials)

and then: 

```
bin/console bolt:setup
```

And in the last step, you said `no` to adding the fixtures.. 


If I do this, it get an empty bolt backend as expected

![Schermafbeelding 2020-08-19 om 14 16 52](https://user-images.githubusercontent.com/1833361/90633734-dd709900-e226-11ea-83a8-d366e6508007.png)

But no error when trying to create a new page: 

![Schermafbeelding 2020-08-19 om 14 17 02](https://user-images.githubusercontent.com/1833361/90633755-e497a700-e226-11ea-8a62-b8fc371b72d0.png)


Any idea what I might do different than you did?

**Comment by antiftw** [comment]
Hi @bobdenotter 

Almost, I think I did not do the setup but manually created the database and schema, with the doctrine commands, and created an admin user myself, because I thought that the setup also inserted the doctrine fixtures.

**Comment by antiftw** [comment]
Hi @bobdenotter,

I wanted to let you know that I tried a clean install, doing it using setup and skipping the last step like you said, and now I am able to create a new page as expected.

After looking into the SetupCommand, to try and find out what could've resulted in the error before, I am confused why this might have happened, because it doesn't really do much more than the three steps. In fact, when you skip the fixtures step the only thing it does extra is generate a new secret, so that's not relevant to the error that occurred.

**Comment by bobdenotter** [comment]
Hi, 

> I am confused why this might have happened, because it doesn't really do much more than the three steps.

Correct, it's pretty much the same, but it allows people who want this to have more control over what they're doing. The results should be the same, though. 

Too bad you also couldn't replicate it, though.. I'll leave the issue open for the time being, and if other people bump into this as well, we can always dig deeper.

**Comment by Wieter** [comment]
I also just encountered this exact same issue (although in prod env). Indeed no page exists, nor textures are loaded. Although I was able to set the `homepage` initially, once I tried adding a normal page this error came.

When I clear the cache using `php bin/console cache:clear` (and subsequently **set the permissions of ./var back** to their correct values!), I can load the `https://www.hidden.hide/bolt/new/pages` exactly 1 time. Once I refresh: boom, 500. 

relevant info from `/www # tail -f var/log/prod.log`:
```
[2020-10-30T20:03:08.393786+00:00] php.CRITICAL: Uncaught Error: Argument 1 passed to Bolt\Twig\ContentExtension::getListTemplates() must be an instance of Bolt\Entity\Field\TemplateselectField, instance of Bolt\Entity\Field given, called in /usr/local/www/apache24/data/www.hidden.hide/www/var/cache/prod/twig/5d/5d64d36b7da1df74af71f41481b777da402f1f3e77046442e8ff0c8196e82bf6.php on line 42 {"exception":"[object] (TypeError(code: 0): Argument 1 passed to Bolt\\Twig\\ContentExtension::getListTemplates() must be an instance of Bolt\\Entity\\Field\\TemplateselectField, instance of Bolt\\Entity\\Field given, called in /usr/local/www/apache24/data/www.hidden.hide/www/var/cache/prod/twig/5d/5d64d36b7da1df74af71f41481b777da402f1f3e77046442e8ff0c8196e82bf6.php on line 42 at /usr/local/www/apache24/data/www.hidden.hide/www/vendor/bolt/core/src/Twig/ContentExtension.php:480)"} []
[2020-10-30T20:03:08.402858+00:00] request.CRITICAL: Uncaught PHP Exception TypeError: "Argument 1 passed to Bolt\Twig\ContentExtension::getListTemplates() must be an instance of Bolt\Entity\Field\TemplateselectField, instance of Bolt\Entity\Field given, called in /usr/local/www/apache24/data/www.hidden.hide/www/var/cache/prod/twig/5d/5d64d36b7da1df74af71f41481b777da402f1f3e77046442e8ff0c8196e82bf6.php on line 42" at /usr/local/www/apache24/data/www.hidden.hide/www/vendor/bolt/core/src/Twig/ContentExtension.php line 480 {"exception":"[object] (TypeError(code: 0): Argument 1 passed to Bolt\\Twig\\ContentExtension::getListTemplates() must be an instance of Bolt\\Entity\\Field\\TemplateselectField, instance of Bolt\\Entity\\Field given, called in /usr/local/www/apache24/data/www.hidden.hide/www/var/cache/prod/twig/5d/5d64d36b7da1df74af71f41481b777da402f1f3e77046442e8ff0c8196e82bf6.php on line 42 at /usr/local/www/apache24/data/www.hidden.hide/www/vendor/bolt/core/src/Twig/ContentExtension.php:480)"} []
[2020-10-30T20:03:08.427742+00:00] php.INFO: User Deprecated: Since pagerfanta/pagerfanta 2.4: The "Pagerfanta\Adapter\DoctrineORMAdapter" class is deprecated and will be removed in 3.0. Use the "Pagerfanta\Doctrine\ORM\QueryAdapter" class from the "pagerfanta/doctrine-orm-adapter" package instead. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Since pagerfanta/pagerfanta 2.4: The \"Pagerfanta\\Adapter\\DoctrineORMAdapter\" class is deprecated and will be removed in 3.0. Use the \"Pagerfanta\\Doctrine\\ORM\\QueryAdapter\" class from the \"pagerfanta/doctrine-orm-adapter\" package instead. at /usr/local/www/apache24/data/www.hidden.hide/www/vendor/pagerfanta/pagerfanta/src/Adapter/DoctrineORMAdapter.php:7)"} []
[2020-10-30T20:03:08.497779+00:00] security.DEBUG: Stored the security token in the session. {"key":"_security_main"} []
```

**Comment by Wieter** [comment]
Sidenote: in `prod` environment the command `php bin/console doctrine:fixtures:load` does not exist. Doctrine fixtures load is only valid in a `dev` env.

**Comment by Wieter** [comment]
A quick-fix for who wants to rush into `production` without fixtures, is the following:

1. perform the usual composer install
2. set up a valid .env.local with `APP_ENV=dev` (because the fixtures are not available for `prod` env)
3. run `php bin/console doctrine:schema:create`
4. run `php bin/console doctrine:fixtures:load` 
5. change the .env.local to contain the desired `APP_ENV=prod`
6. clear loaded static data by running `rm -r ./public/files/{animal,people,stock}`
7. drop the database structure, run `php bin/console doctrine:schema:drop --force`
8. re-initialize the database, run `php bin/console doctrine:schema:create`
9. add superuser, run `php bin/console bolt:add-user --admin`
10. ensure default themes are present, run `php bin/console bolt:copy-themes`
11. ensure proper permissions as described [here](https://docs.bolt.cm/4.0/installation/permissions) (personally I additionally had to chown the config,var,public/files,public/theme,public/thumbs dirs to www)
 
Basically this loads and purges the fixtures. Not very fancy but it worked perfectly for my needs.

**Comment by bobdenotter** [comment]
@Wieter If you don't wish the fixtures to fetch images, you could perhaps also run `doctrine:fixtures:load --no-interaction --group=without-images` ?

**Comment by Wieter** [comment]
> @Wieter If you don't wish the fixtures to fetch images, you could perhaps also run `doctrine:fixtures:load --no-interaction --group=without-images` ?

@bobdenotter Oh, yes. How did I miss that one? (note to self: need to enhance caffeine intake.. ). I'll edit above post accordingly.
On a side-note, during all things above I might have encountered a security issue. As suggested there, I sent an email yesterday. However, I think the stated email address there is incorrect, as I received the following:
```
Final-Recipient: rfc822;security@bolt.cm
Action: failed
Status: 5.5.0
Diagnostic-Code: smtp;550 No such recipient here
```
Where can I forward this issue to? 
Perhaps here publicly is not too bad actually, because it concerns a configuration step during setup.
